cli/args/mod.rs

```diff
@@ -1460,29 +1460,36 @@ impl CliOptions {
   }
 
   pub fn watch_paths(&self) -> Vec<PathBuf> {
-    let mut paths = if let DenoSubcommand::Run(RunFlags {
+    let mut full_paths = Vec::new();
+    if let DenoSubcommand::Run(RunFlags {
       watch: Some(WatchFlagsWithPaths { paths, .. }),
       ..
     }) = &self.flags.subcommand
     {
-      paths.clone()
-    } else {
-      Vec::with_capacity(2)
+      for path in paths {
+        let converted_path = PathBuf::from(path);
+        if converted_path.is_absolute() {
+          full_paths.push(converted_path);
+        } else {
+          full_paths.push(self.initial_cwd().join(path));
+        }
```

Comment:
```suggestion
        full_paths.extend(paths.into_iter().map(|path| self.initial_cwd().join(path));
```

I think `.join` already handles the `is_absolute` logic https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.join

Reply:
Good catch, thanks for spotting that!